### PR TITLE
Prepare for contributions

### PR DESCRIPTION
**Pull request by bennyz**
- add scripts for running tests and creating the database
- run database migrations on server startup
- add README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# qarax
+qarax aims to be a firecracker orchestrator, it is currently in very early stages
```
Comment (matanshukry) [inline README.md:2]: Explain what Firecracker is shortly to those who aren't familiar to it, and even add link to it for more info

File: README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+### Development
+
+#### Create the database
+```shell
+./create_db.sh
```
Comment (matanshukry) [inline README.md:8]: File is in root? perhaps add script directory?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+#### Run the server
+```shell
+$ cargo run
```
Comment (matanshukry) [inline README.md:13]: How to install cargo? is the package name cargo? link to their site if too lazy? ..

File: README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+#### Execute tests
+```shell
+./run_tests.sh
+```
+Note: a simple `cargo test` will not work as the tests require a database and cannot be run in parallel.
```
Comment (matanshukry) [inline README.md:20]: How is parallel related exactly? what does run_tests does to parallelism?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+```json
+{"host_id":"365e5061-62b9-41e5-9766-47fcd2c51721"}
+```
+
+Installing a host:
+```shell
+$ curl -XPOST http://localhost:8000/hosts/365e5061-62b9-41e5-9766-47fcd2c51721/install -d '{ "local_node_path": "/path/to/qarax-node", "fcversion": "v0.21.1" }' -H "Content-Type: application/json"
+```
+
+Note: The source of [qarax-node](https://github.com/qarax/qarax-node) is required
```
Comment (matanshukry) [inline README.md:38]: Requires where? how?

File: README.md (hunk 3 above)
Comment (bennyz) [reply README.md:13]: cargo is rust's package manager, I'll add a prerequisites section

File: README.md (hunk 5 above)
Comment (bennyz) [reply README.md:38]: it has to reside in `/path/to/qarax-node`

File: README.md (hunk 4 above)
Comment (bennyz) [reply README.md:20]: see https://github.com/qarax/qarax/pull/36/files#diff-c08082f308f76bbde9882b88503605beR8
`RUST_TEST_THREADS=1` forces it to run with a single thread

File: README.md (hunk 1 above)
Comment (bennyz) [reply README.md]: Added a link

File: README.md (hunk 2 above)
Comment (bennyz) [reply README.md:22]: it will cause issues with running the tests, but I will change it in the future

Comment (bennyz) [reply README.md]: never mind, changed it now

**Pull request merged**